### ISSUE TITLE
great job!请教训练的问题

### ISSUE THREAD

**Issue by gangxu822** [bcmi/GPSDiffusion-Object-Shadow-Generation-SDXL#12]
很不错的工作，但是训练时间有点长：07/08/2025 09:52:08 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 4
07/08/2025 09:52:08 - INFO - __main__ -   Gradient Accumulation steps = 1
07/08/2025 09:52:08 - INFO - __main__ -   Total optimization steps = 347800
 Loading prompt embeddings from cache...
Steps:   0%|          | 701/347800 [11:48<95:05:59,  1.01it/s, loss=0.199, lr=1e-5]  
这是要95个小时？一般可以改哪些配置缩短这个时间？你们一般训练要多久

**Comment by nononononno** [comment]
accelerate config, 运行之后在里面选择多卡训练，可以参阅accelerate的官方文档

**Comment by gangxu822** [comment]
> accelerate config, 运行之后在里面选择多卡训练，可以参阅accelerate的官方文档

多谢~

**Comment by nononononno** [comment]
> }, os.path.join(save_dir, "ip_adapter.ckpt")) controlnet.save_pretrained(os.path.join(save_dir, "controlnet")) 参照代码： 最后只会输出ip_adapter和controlnel的checkpoint，test时候用这两个新训出来的，其他的模型不变？

对，只训练controlnet和ip-adapter,其他部分参数冻结

**Comment by gangxu822** [comment]
训出来ip_adapter.ckpt大小差了一倍（一个600多m，一个1.3g），这正常吗？
  ll -rt ckpts_GPS_sdxl/epoch-22/ip_adapter.ckpt 
-rw-r--r-- 1 root root 698429518 Jul  8 17:52 ckpts_GPS_sdxl/epoch-22/ip_adapter.ckpt
 ll -rt pretrained_models/ip_adapter.ckpt 
-rw-rw-r-- 1 root root 1380003918 Jun 27 21:49 pretrained_models/ip_adapter.ckpt

**Comment by nononononno** [comment]
> 训出来ip_adapter.ckpt大小差了一倍（一个600多m，一个1.3g），这正常吗？ ll -rt ckpts_GPS_sdxl/epoch-22/ip_adapter.ckpt -rw-r--r-- 1 root root 698429518 Jul 8 17:52 ckpts_GPS_sdxl/epoch-22/ip_adapter.ckpt ll -rt pretrained_models/ip_adapter.ckpt -rw-rw-r-- 1 root root 1380003918 Jun 27 21:49 pretrained_models/ip_adapter.ckpt

最好看下测试效果

**Comment by gangxu822** [comment]
> > }, os.path.join(save_dir, "ip_adapter.ckpt")) controlnet.save_pretrained(os.path.join(save_dir, "controlnet")) 参照代码： 最后只会输出ip_adapter和controlnel的checkpoint，test时候用这两个新训出来的，其他的模型不变？
> 
> 对，只训练controlnet和ip-adapter,其他部分参数冻结。如果你训练的是shadow之外的task并且效果一般的话，可以考虑重新训练geometry encoder，可以参考GPSDiffusion项目中的train_bbx.py和train_cls.py

需要从你的Pretrain model开始训吗？如果需要从你的pretrain模型开始训，配置怎么设？
